Automattic/wp-calypso
# Step-by-step credentials verification (progress only)

* [context and past patches](https://wp.me/pbuNQi-GM)
* [#47898 (original data-layer changes, now reverted)](https://github.com/Automattic/wp-calypso/pull/47898)
* **you are here:** [#47614 (i4 verification screen with progress only)](https://github.com/Automattic/wp-calypso/pull/47614)
* [#48106 (full i4 verification screen)](https://github.com/Automattic/wp-calypso/pull/48106)
* [#48492 (data-layer with wpcom-proxy-request fix)](https://github.com/Automattic/wp-calypso/pull/48492)

#### Changes proposed

This patch replaces the “Testing connection…” notice below with an early version of the i4 step-by-step verification screen, using the streamed progress messages that #47898 made available.

![screenshot](https://user-images.githubusercontent.com/465303/101018585-d4ca8580-35bf-11eb-9989-e46431cf96e7.png)

Successes and failures work the same way for now (notice and return to credentials form). #48106 introduces their i4-based replacement, along with the error handling improvements it relies on.

These changes affect the “advanced credentials” flow, which is currently green only, so blue should be unaffected.

#### Testing instructions

Check out this patch:

```
$ git fetch # or clone if necessary
$ git checkout add/credentials-testing-progress
```

Update your node_modules, then spin up green locally:

```
$ yarn
$ yarn start-jetpack-cloud
```

Open green and choose a Rewind-enabled site, then click **Settings**.

Verify that you can save working credentials successfully, with progress displayed similarly to the screen recording below, followed by the usual notice:

![screenshot](https://user-images.githubusercontent.com/465303/101018489-afd61280-35bf-11eb-8d69-15cae85c787e.png)

Now try to save the same credentials except for an incorrect password or private key, and verify that progress is displayed similarly to the screen recording below, followed by the usual notice:

![screenshot](https://user-images.githubusercontent.com/465303/101018676-f3308100-35bf-11eb-8d90-552e8dd1e7ec.png)

#### Screen recording (click to play)

[![screenshot from video](https://user-images.githubusercontent.com/465303/101015703-e6aa2980-35bb-11eb-9433-69706f8735f1.png)](https://bucket.daz.cat/0490e7e7566c9f00.mp4)
